src/puppet-wechat.ts

```diff
@@ -317,9 +318,16 @@ export class PuppetWeChat extends PUPPET.Puppet {
   }
 
   override async messageFile (messageId: string): Promise<FileBoxInterface> {
-    const rawPayload = await this.messageRawPayload(messageId)
-    const fileBox    = await this.messageRawPayloadToFile(rawPayload)
-    return fileBox
+    do {
+      const rawPayload = await this.messageRawPayload(messageId)
+      switch (rawPayload.MMStatus) {
+        case WebMessageStatus.SENDING:
+          await new Promise((resolve) => setTimeout(resolve, 50))
```

Comment:
How can we know the `SENDING` will be finished after 50ms?

Is there a better way, for example, an event to let us know the `SENDING` has been finished?

Reply:
`50ms` is the polling frequency, it runs in a loop.
I have another solution, using `setter` on `MsgId` property to hook the `MMStatus` change event, but I can't make sure it works stably.

Reply:
If I'm understanding correctly, the `50ms` is the long polling interval of the webwx-app itself? Could you please help me to point out the related code in the webwx-app?

And if so, I think the current solution in the PR is acceptable by adding some comments to explain the `50ms` pooling interval with the related source code link (the [webwx-app-tracker](https://github.com/wechaty/webwx-app-tracker) link will be recommended).

After the comments have been added, we can merge this PR first and keep trying to improve it in the future.

Reply:
Unfortunately it is a custom interval, plz don't merge now, let me try `root:msgSend:success` event.

Reply:
Great, I agree that the `root:msgSend:success` event will be the solution to this workaround.

Looking forward to your update, thank you very much!

Reply:
`root:msgSend:success` event only emit on message success, await it may cause deadlock when message fails, but I found `$watch` to implement it.